### ISSUE TITLE
Caching

### ISSUE THREAD

**Issue by tkbits9** [TomFahey/nginx-cloudflare-jwt-checker#4]
Hi @TomFahey 

Great work there mate. 

Are you using this in prod at the mom? Also as full disclosure, I'm not a C expert so apologies if this is a silly question but where is the caching for the validation?

Many thanks,

**Comment by TomFahey** [comment]
Hi @tkbits9,

Thanks very much, I hope people find it useful!

I'm no expert in C, nginx or web security either, so no worries there. I'm not sure what you mean by "where is the caching for the validation?"?

**Comment by tkbits9** [comment]
yes, it was really useful. I'm still exploring this option since the alternative is to deal with certificates or offload that to the app (which I don't want to do).

My point with caching was around the validation of the token to prevent repeat validations on the same token (assuming it's still valid)?

**Comment by TomFahey** [comment]
Great to hear!

I'm afraid there is no caching at the moment - every request with a token is validated currently; I could take a look at trying to add this as a feature, however that would raise the issue of how long to actually keep a cached token for, as token expiry is actually subject to quite a number of different factors, such as manual revocation by the user, rotation of the public certs by Cloudflare and the choice of session duration in the access policy for that particular endpoint (which may be completely different to another endpoint, under the same domain). So there wouldn't necessarily be an easy way to decide whether a cached token should be expired or not...

I do agree that it is wasteful to check every single request though, especially when they're being received immediately after one another - this is one of the things that actually bugged me with the sample code that Cloudflare provided in their documentation [here](https://developers.cloudflare.com/cloudflare-one/identity/users/validating-json). Perhaps there could be a happy middle-ground though.

**Comment by tkbits9** [comment]
yes, caching is hard everywhere :) But I think we could explore an expiration variable to maintain flexibility so each request would only be checked against a new dataset if the cache expired!
